ably/proto_message_decoding_test.go

```diff
@@ -0,0 +1,147 @@
+package ably
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/ably/ably-go/ably/internal/ablyutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type fixture struct {
+	Data          string `json:"data"`
+	Encoding      string `json:"encoding"`
+	ExpectedType  string `json:"expectedType"`
+	ExpectedValue any    `json:"expectedValue"`
+}
+
+func loadFixtures() ([]fixture, error) {
```

Comment:
There is some similar code for loading the crypto fixtures in the `ablytest` package [here](https://github.com/ably/ably-go/blob/main/ablytest/cryptodata.go), is it worth being consistent with that existing code?

Reply:
Both functions read test fixture data. But the data does have different schemas, and different encodings.
And LoadCryptoData() returns different data types. 
DNRY is a good principle. But here, a I think in this case, and in loadMsgpackFixtures, a bit of repetition leads to simpler and more understandable code

Reply:
Ok sorry, I didn't mean use that exact method, but perhaps have `ablytest.LoadMessageEncodingFixtures` and `ablytest.LoadMsgpackFixtures` (i.e. put all the test fixture loading code in the same place), but I understand if you don't think that's useful.

Reply:
I could do that if you want. But it is a trivial function, so I think it is probably better to have it not exported and defined nearer the test that uses it.

Reply:
👍